uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -90,7 +125,14 @@ define([
          * @return {Promise} Resolved with {Object} Updated Leave request
          */
         update: function () {
-          return LeaveRequestAPI.update(this.toAPI());
+          var comments = this.comments;
```

Comment:
Initially I though of doing the same, but we need to save the comments in an temporary variable (`comments`) so that it can be reattached to the object once the promise for the API call is resolved. That's why went with this approach.